pkg/plugins/manager/loader/loader.go

```diff
@@ -25,41 +26,106 @@ var _ plugins.ErrorResolver = (*Loader)(nil)
 
 type Loader struct {
 	pluginFinder        finder.Finder
-	processManager      process.Service
 	pluginRegistry      registry.Service
-	roleRegistry        plugins.RoleRegistry
-	pluginInitializer   initializer.Initializer
 	signatureValidator  signature.Validator
 	signatureCalculator plugins.SignatureCalculator
 	assetPath           *assetpath.Service
 	log                 log.Logger
 	cfg                 *config.Cfg
 
+	hooksRegistry hooks.Registry
+	hooksRunner   hooks.Runner
+
 	errs map[string]*plugins.SignatureError
 }
 
-func ProvideService(cfg *config.Cfg, license plugins.Licensing, authorizer plugins.PluginLoaderAuthorizer,
-	pluginRegistry registry.Service, backendProvider plugins.BackendFactoryProvider, pluginFinder finder.Finder,
-	roleRegistry plugins.RoleRegistry, assetPath *assetpath.Service, signatureCalculator plugins.SignatureCalculator) *Loader {
-	return New(cfg, license, authorizer, pluginRegistry, backendProvider, process.NewManager(pluginRegistry),
-		roleRegistry, assetPath, pluginFinder, signatureCalculator)
+func ProvideService(cfg *config.Cfg, authorizer plugins.PluginLoaderAuthorizer,
+	pluginRegistry registry.Service, pluginFinder finder.Finder, assetPath *assetpath.Service,
+	signatureCalculator plugins.SignatureCalculator, hooksRegistry hooks.Registry, hooksRunner hooks.Runner,
+	pluginInitializer initializer.Initializer,
+
+	// TODO: hooks: Provided just for hooks side-effects, find a better way
+	_ process.Service,
+) *Loader {
+	return New(cfg, authorizer, pluginRegistry, assetPath, pluginFinder, signatureCalculator, pluginInitializer, hooksRegistry, hooksRunner)
 }
 
-func New(cfg *config.Cfg, license plugins.Licensing, authorizer plugins.PluginLoaderAuthorizer,
-	pluginRegistry registry.Service, backendProvider plugins.BackendFactoryProvider,
-	processManager process.Service, roleRegistry plugins.RoleRegistry,
-	assetPath *assetpath.Service, pluginFinder finder.Finder, signatureCalculator plugins.SignatureCalculator) *Loader {
+func New(cfg *config.Cfg, authorizer plugins.PluginLoaderAuthorizer,
+	pluginRegistry registry.Service,
+	assetPath *assetpath.Service, pluginFinder finder.Finder, signatureCalculator plugins.SignatureCalculator,
+	pluginInitializer initializer.Initializer,
+	hooksRegistry hooks.Registry, hooksRunner hooks.Runner) *Loader {
+	logger := log.New("plugin.loader")
+
+	// TODO: hooks: Move those to separate services
+	hooksRegistry.RegisterBeforeLoadHook(func(ctx context.Context, plugin *plugins.Plugin) error {
+		if plugin.IsApp() {
+			setDefaultNavURL(plugin)
+		}
+		return nil
+	})
+	hooksRegistry.RegisterBeforeLoadHook(func(ctx context.Context, plugin *plugins.Plugin) error {
+		if plugin.Parent != nil && plugin.Parent.IsApp() {
+			configureAppChildPlugin(plugin.Parent, plugin)
+		}
+		return nil
+	})
```

Comment:
I'm still not 100% sure if we should implement those as hooks, because they are logically related to "loading" a plugin, so it makes sense for them to be in the plugin "loader"